### DISCUSSION TITLE
Porous media sinks

### DISCUSSION THREAD

**Discussion by MatiasCastilloH** [idaholab/moose#16423] [Q&A General] [answered]
Dear MOOSE,

I am trying to perform a numerical model of groundwater extraction. I have a set up with several extraction wells and a river. The idea is to asses the interference between the wells (overlapping of well cones) and the interaction that they have with the river. Also, at some point, the level of the river increases and a contaminant is injected in it. Here, the objective is to asses whether the contaminant get into the pumped water by the wells. 

Is possible to perform such models with MOOSE? 
If so, is it a correct/realistic description of the wells the "point source sink"? As, for example, the Theis example displayed on the groundwater models example page? Seems the most reasonable approach to simulate a cone and I also have external data on pumping rates so I can combine it with PorousFlowPointSourceFromPostprocessor.

Thanks and sorry for the simple question, I am new with MOOSE
Matias Castillo

**Comment by permcody** [comment]
Tag @WilkAndy

**Comment by josebastiase** [comment]
Hi Matias,

Sounds like a cool simulation! I think that is possible to simulate that with MOOSE. I’m not sure if there are examples for contaminant transport though. Probably Andy can comment on that.

Regarding your extraction well set up, as Maelle and Andy nicely demonstrated at the end of the 'groundwater examples'  is always better to use boreholes. Plus, is often good to document a Theis example since is useful as verification of a well-known problem, but be aware that in that case the aquifer is saturated so is not really a ‘cone drawdown’. 

Setting up a borehole in Porous Flow is quite simple, I suggest you to take a look at bh02.i and bh03.i tests [here](https://mooseframework.inl.gov/source/dirackernels/PorousFlowPeacemanBorehole.html). Just be aware that you will have to set porepressure instead of fluxes in your MOOSE input file.

Cheers!
Jose

**Comment by WilkAndy** [comment, marked as answer]
Hi Matias,

Yes, this is exactly the sort of thing that can be done with PorousFlow.  I've run almost exactly this sort of simulation in the past (for me, the boreholes were pumping the contaminant, and the stream was receiving it).  As mentioned in the groundwater doco, setting up your mesh will turn out to be the most difficult thing.

Whether you use just a single point for your well depends on the resolution of your problem.  You clearly don't need more than 1 point per finite element, since the finite element size dictates the resolution of the problem, but if your wells span multiple finite elements you'll probably want to model them using multiple points (just include more points in the borehole's geometry file).

Yes, you can simply use `PorousFlowPointSourceFromPostprocessor` and prescribe a pumping rate.  The only thing you have to be careful with is prescribing an unrealistically high pumping rate (compared with the aquifer hydraulic conductivity) and then the hydraulic heads or porepressures become unrealistically negative (corresponding to desaturation of the aquifer).

Yes, you can include contaminants: you will utilize the `mass_fraction_vars` input if you are using one of the PorousFlow Actions.  I suggest you start by reading this tutorial page:  https://mooseframework.inl.gov/modules/porous_flow/tutorial_06.html .    Then, you need to read the "numerical diffusion" page: https://mooseframework.inl.gov/modules/porous_flow/numerical_diffusion.html  .   Perhaps you don't need to understand everything on that page, but you certainly need to appreciate the concept of numerical diffusion, because it is likely to impact your "break-through" times and your interpretation of your results.  That knowledge will be valuable, even if you judge that no stabilization is needed for your problem (I would start with no stabilization and see if your results are adequate, since that's the easiest).

Regarding the contaminants, if you issue this command in the moose/porous_flow directory
```
find . -name "*.i" | xargs grep mass_fraction_vars
```
you will see a lot of tests that you can follow, in addition to the ones mentioned in the above doco pages.

Finally, are you willing to share your results?  It's hard to tell, but they might be a useful addition to the "Groundwater Modelling" page that Maelle and I wrote.

andy

**Comment by MatiasCastilloH** [comment]
Hi,
@josebastiase 
Thanks for your reply and for pointing out that a cone drawdown doesn't occur (I'm still getting use to the concept of porepressure). Thanks also for the borehole examples, that's really helpful! Saludos :+1: 

@WilkAndy 
I would be happy to share the results plus the model with the community once is finished. I still need to generate a realistic mesh, but for now I just have a quadratic one with three layers, that makes the job.

The first step would be running a steady state simulation with the pumps and the river and asses the interaction among them. As I understand, there are two options to set the wells:

1.- PorousFlowPointSourceFromPostprocessor - I can set up flows from external file but is not as realistic as a borehole. Further, it has to be set at a node of the mesh.

2.- PorousFlowPeacemanBorehole - More realistic and it can be propositioned anywhere in the mesh. However, there is not direct way to control the pumped flow. This has to be tuned by setting a 'bottom' porepressure.

Is that correct?
To asses the interaction among wells, how can I obtain the capture zone of the wells (streamlines)?

Thanks for the quick replies!
Matias Castillo

**Comment by WilkAndy** [comment]
What you write about `PorousFlowPeacemanBorehole` is correct.

What you write about `PorousFlowPointSourceFromPostprocessor` is not really correct.  You may position it anywhere in the mesh, and it withdraws the amount of water specified in the postprocessor.  You just must be careful with interpreting the results.  Let's say you position the borehole points so they lie directly in the middle of finite elements.  Then the water withdrawal will occur equally from the water at each of the nodes of those finite elements: it's as if you had an enormous element-sized borehole that abstracts water.  This is not *wrong* - it's just a reflection of the fact that your resolution in a finite-element model is set by the element size.  On the other hand, if you position the borehole points so they lie exactly on nodes then the water withdrawal will occur from those nodes only.  This does not make your simulation significantly more accurate, since your resolution is still set by the element size - it's just easier to convince yourself that the results are actually more accurate!

Regarding the streamlines - you could use a `PorousFlowDarcyVelocityComponent` AuxVariable and plot the flow directions in paraview.
